heat/core/linalg.py

```diff
@@ -63,7 +63,7 @@ def transpose(a, axes=None):
 
         return dndarray.DNDarray(transposed_data, transposed_shape, a.dtype, transposed_split, a.device, a.comm)
     # if not possible re- raise any torch exception as ValueError
-    except RuntimeError as exception:
+    except Exception as exception:
```

Comment:
This is a dangerous bit of code. what about `except (RuntimeError, IndexError) as exception` ? this is also dangerous but is a bit more narrow. I asked for @Markus-Goetz to review this PR as well to see if he can see the problem.

However, I am still not sure about why the docker is throwing an IndexError here in the first place. all of my local tests do not show this behavior.

Reply:
Catching Exception generally is not something we should do here. The proposed solution of Daniel is much better. However, I would like to know why there is a difference in exceptions to begin with? Is it simply a PyTorch version thing?